From 6709342d25629debcab15adbd16d7d08948e2c9e Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@gnome.org>
Date: Thu, 10 Nov 2011 23:57:03 +0100
Subject: [PATCH] GtkTrayIcon:fg-color is a GdkRGBA

... and not a GdkColor. So use the right variable, and the right free function.
Fixes a crash on session startup in gnome-settings-daemon, when using the gnome3
fallback session.

https://bugzilla.gnome.org/show_bug.cgi?id=663816
---
 libgnomekbd/gkbd-status.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libgnomekbd/gkbd-status.c b/libgnomekbd/gkbd-status.c
index 2f5b41f..10c95bc 100644
--- a/libgnomekbd/gkbd-status.c
+++ b/libgnomekbd/gkbd-status.c
@@ -126,7 +126,7 @@ static void
 gkbd_status_render_cairo (GkbdStatusPrivHack * gkh, cairo_t * cr, int group)
 {
 	double r, g, b;
-	GdkColor *fg_color;
+	GdkRGBA *fg_color;
 	gchar *font_family;
 	int font_size;
 	PangoFontDescription *pfd;
@@ -155,7 +155,7 @@ gkbd_status_render_cairo (GkbdStatusPrivHack * gkh, cairo_t * cr, int group)
 
 	g_object_get (gkh->tray_icon, "fg-color", &fg_color, NULL);
 	cairo_set_source_rgb (cr, fg_color->red, fg_color->green, fg_color->blue);
-	gdk_color_free (fg_color);
+	gdk_rgba_free (fg_color);
 
 	gkbd_indicator_config_get_font_for_widget (ind_cfg,
 						   gkh->tray_icon,
-- 
1.7.8

